### PR TITLE
feat: add usd value of donation during checkout

### LINKED ISSUE #1537: Add value of donation indicator
See [Loom](https://www.loom.com/share/e9f95926a61643c6a4ede3624f7810e6) overview

**User Story:**
As a donor
I want to know what the USD value of my contributions are
So that I know if they meet the minimum donation threshold for the round

**Acceptance Criteria**
GIVEN that I have added my donation amount in ETH/DAI 
WHEN I look at my cart overview 
THEN I see the USD value of my contribution per project 

GIVEN that I am ready to checkout 
WHEN I look at my total donation amount 
THEN I see the USD value of my total donation amount 

**Product & Design Links:**
* [Figma](https://www.figma.com/file/cgsZMTG71f2Hf89J38simK/Explorer-S17?node-id=526-14125&t=G1DcgMmtQmMV2KSE-0)

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by bhargavaparoksham**
##### Description

As a donor
I want to know what the USD value of my contributions are
So that I know if they meet the minimum donation threshold for the round

##### Refers/Fixes

fixes #1537

### CODE REVIEW

File: packages/grant-explorer/src/features/api/utils.ts [hunk 1]
```diff
@@ -420,3 +421,50 @@ export const listenForOutsideClicks = ({
     });
   };
 };
+
+export const useTokenPrice = (tokenId: string | undefined) => {
+  const [tokenPrice, setTokenPrice] = useState<number>();
+  const [error, setError] = useState<Response | undefined>();
+  const [loading, setLoading] = useState(false);
+
```
Comment (hmrtn) [inline packages/grant-explorer/src/features/api/utils.ts:429]: imo I think checking for an empty token id is better than running through the request. 
```
  if (!tokenId) {
    setError(new Error('Token ID is not provided.'));
    return;
  }
```

**Comment by michellema1208** [comment]
@thelostone-mc @bhargavaparoksham it looks like the build failed, so I'm unable to test on PM review side. Can you tag me when those builds have passed/deploy links are ready and I'll come back and test?

File: packages/common/src/index.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { useParams } from "react-router";
 import useSWR from "swr";
 import { isAddress } from "viem";
+import { useMemo, useState } from "react";
+import redstone from 'redstone-api';
```
[the remark this answers is not in the document]
Comment (thelostone-mc) [reply packages/common/src/index.ts]: donee

**Comment by thelostone-mc** [comment]
@michellema1208 It's good for review

File: packages/common/src/index.ts [hunk 3]
```diff
@@ -355,3 +356,56 @@ export const getUTCDate = (date: Date): string => {
...
+  if (!tokenId) return {
+    data: 0,
+    error,
+    loading,
+  };
+
+  useMemo(async () => {
+    setLoading(true);
+
+    const tokenPriceEndpoint = `https://api.redstone.finance/prices?symbol=${tokenId}&provider=redstone&limit=1`;
```
Comment (hmrtn) [inline packages/common/src/index.ts:381]: should this be an env?

**Comment by thelostone-mc** [comment]
> Is there a motive for dropping coingecko entirely? 
Would it be appropriate to use it as a fallback rather than remove it entirely?

Coinbase keeps failing on the frontend constantly. 
Redstone on the other hand seems more reliable for the frontend. 
Fallbacks I'd say maybe we add it if need do.

File: packages/common/src/index.ts (hunk 3 above)
Comment (thelostone-mc) [reply packages/common/src/index.ts:381]: That URL doesn't change. Hence didn't add it as an env

**Review by michellema1208 — APPROVED** [review]
LGTM!

**Review by melissa-neira — APPROVED** [review]

**Pull request merged**